test/test_browser.py

```diff
@@ -699,15 +699,20 @@ def setup(assetLocalization):
           <center><canvas id='canvas' width='256' height='256'></canvas></center>
           <hr><div id='output'></div><hr>
           <script type='text/javascript'>
-            window.onerror = (error) => {
+            const handler = event => {
+              event.stopImmediatePropagation();
+              const error = String(event instanceof ErrorEvent ? event.message : (event.reason || event));
               window.disableErrorReporting = true;
               window.onerror = null;
               var result = error.includes("test.data") ? 1 : 0;
               var xhr = new XMLHttpRequest();
               xhr.open('GET', 'http://localhost:8888/report_result?' + result, true);
               xhr.send();
+              console.log(event, error);
               setTimeout(function() { window.close() }, 1000);
             }
+            window.addEventListener('error', handler);
+            window.addEventListener('unhandledrejection', handler);
```

Comment:
Is this change needed because we now generate `unhandledrejection` in more cases than before?

Reply:
Throwing an error inside a promise, or rejecting for any other reason will generate ONLY an `unhandledrejection` event, not an `error` event. Unless of course you `catch()` it.